packages/arb-token-bridge-ui/src/hooks/useError.ts

```diff
@@ -0,0 +1,173 @@
+import { useCallback } from 'react'
+import * as Sentry from '@sentry/react'
+import { isUserRejectedError } from '../util/isUserRejectedError'
+import { useAccount } from 'wagmi'
+import { useNetworks } from './useNetworks'
+
+/**
+ * Categories for classifying errors
+ */
+export type ErrorCategory =
+  | 'token_validation'
+  | 'token_approval'
+  | 'token_transfer'
+  | 'contract_interaction'
+  | 'contract_validation'
```

Comment:
i think we should add `claim` and `allowance_check` too and only have these two as catchalls